### PR TITLE
HPO & Fulldata

### PR DESCRIPTION
**Pull request by MARMOTatZJU**
____Major changes____

* Add hyper-parameter search script
  * _main/hpo.py_
  * _experiment_
  * _videoanalyst/utils/hpo.py_
* Add full training data
  * coco / det / vid / trackingnet / lasot / got10k
  * with caching mechanism (reducing storage I/O by caching indexes into memory)
  * TODO: perform verification with full data (under batchsize=256)

____Minor changes____

* reorganized experiment / tools (file structure)
  * add hierarchy to avoid too many files under _experiments_ & _tools_ folder

### CODE REVIEW

File: videoanalyst/data/dataset/dataset_impl/det.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+from videoanalyst.data.dataset.dataset_base import TRACK_DATASETS, DatasetBase
+from videoanalyst.evaluation.got_benchmark.datasets import ImageNetVID
+from videoanalyst.pipeline.utils.bbox import xywh2xyxy
+
+_VALID_SUBSETS = ['train', 'val']
+
+
+@TRACK_DATASETS.register
+class DETDataset(DatasetBase):
```
Comment (lzx1413) [inline videoanalyst/data/dataset/dataset_impl/det.py:24]: Rename to another specifial name?

File: videoanalyst/data/dataset/dataset_impl/trackingnet.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        dataset split name (train|val|train_val)
+    ratio: float
+        dataset ratio. used by sampler (data.sampler).
+    max_diff: int
+        maximum difference in index of a pair of sampled frames 
+    check_integrity: bool
+        if check integrity of dataset or not
+    """
+    default_hyper_params = dict(
+        dataset_root="datasets/ILSVRC2015",
```
Comment (lzx1413) [inline videoanalyst/data/dataset/dataset_impl/trackingnet.py:37]: replace with right data path?

Comment (MARMOTatZJU) [reply videoanalyst/data/dataset/dataset_impl/trackingnet.py:37]: It is the right path. ILSVRC2015-DET and ILSVRC2015-VID share the same root & file structure. I consider to give a description of dataset structure in future.

Comment (lzx1413) [reply videoanalyst/data/dataset/dataset_impl/trackingnet.py:37]: But this is trackingnet dataset.

File: videoanalyst/data/dataset/dataset_impl/det.py (hunk 1 above)
Comment (MARMOTatZJU) [reply videoanalyst/data/dataset/dataset_impl/det.py:24]: Done

File: videoanalyst/data/dataset/dataset_impl/trackingnet.py (hunk 2 above)
Comment (MARMOTatZJU) [reply videoanalyst/data/dataset/dataset_impl/trackingnet.py:37]: Fixed

**Comment by MARMOTatZJU** [comment]
Updated.

**Pull request merged**